### PR TITLE
install.sh|upload.sh: Parse arguments without getopts | Improve input fle/folder parsing | Misc

### PR DESCRIPTION
**Pull request by Akianonymus**
e.g: gupload -d -o a b, here a is file/folder, b is gdrive folder.

   : gupload a b, here a is file/folder, b is gdrive folder.

   : gupload -d -o a b -s c d, here a, b and c are file/folder, d is gdrive folder.

* _json_value: Don't check for arguments

    because it works on pipe output too

* _full_path: Improve the function for various usescases, make it more accurate

* fix some typos

* Update README

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -172,10 +170,13 @@ _dirname() {
...
 ###################################################
 _full_path() {
-    case "${1?${FUNCNAME[0]}: Missing arguments}" in
-        /*) printf "%s\n" "${1}" ;;
-        *) printf "%s\n" "${PWD}/${1}" ;;
-    esac
+    [[ $# = 0 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare input="${1}"
+    if [[ -f ${input} ]]; then
+        printf "%s/%s\n" "$(cd $(_dirname "${input}") && pwd)" "${input##*/}"
```
Comment (labbots) [inline install.sh:176]: Codacy Issue found: [Quote this to prevent word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

**Comment by labbots** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879&bid=18269338)

File: install.sh [hunk 2]
```diff
@@ -172,10 +170,13 @@ _dirname() {
...
+    elif [[ -d ${input} ]]; then
+        printf "%s\n" "$(cd ${input} && pwd)"
```
Comment (labbots) [inline install.sh:178]: Codacy Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

File: utils.sh [hunk 3]
```diff
@@ -210,10 +210,13 @@ _extract_id() {
...
-    case "${1?${FUNCNAME[0]}: Missing arguments}" in
-        /*) printf "%s\n" "${1}" ;;
-        *) printf "%s\n" "${PWD}/${1}" ;;
-    esac
+    [[ $# = 0 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare input="${1}"
+    if [[ -f ${input} ]]; then
+        printf "%s/%s\n" "$(cd $(_dirname "${input}") && pwd)" "${input##*/}"
+    elif [[ -d ${input} ]]; then
+        printf "%s\n" "$(cd ${input} && pwd)"
```
Comment (labbots) [inline utils.sh:218]: Codacy Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

File: utils.sh [hunk 4]
```diff
@@ -210,10 +210,13 @@ _extract_id() {
...
 ###################################################
 _full_path() {
-    case "${1?${FUNCNAME[0]}: Missing arguments}" in
-        /*) printf "%s\n" "${1}" ;;
-        *) printf "%s\n" "${PWD}/${1}" ;;
-    esac
+    [[ $# = 0 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare input="${1}"
+    if [[ -f ${input} ]]; then
+        printf "%s/%s\n" "$(cd $(_dirname "${input}") && pwd)" "${input##*/}"
```
Comment (labbots) [inline utils.sh:216]: Codacy Issue found: [Quote this to prevent word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

**Comment by labbots** [comment]
@Akianonymus what is the need to change the use of getopts? if the user wants to upload multiple files then it could be handled by passing multiple -f option. This I believe is the standard for most linux commands. Handling it through unnamed argument doesn't seam to be the right method. So could you be please elaborate the reason for the change.

**Comment by Akianonymus** [comment]
@labbots First, i removed getopts to decrease code lines.

Now about handling arguments, i was trying to match it to advance programs out there. e.g: curl handles multiple arguments like that, although it still have `--next` option for handling multiple arguments.

Note: While i implemented that, the previous behaviour is still there, i.e

gupload filename gdrive_folder someflags

I did this basically to we can handle arguments given between flags, as i have explained in the readme.

To sum up, yes this was basically an attempt to match argument handling like curl, and yes it is convenient to have multiple arguments without using -f option.

I believe there is nothing wrong in having multiple choices within disturbing the existing ones. Anyway, will remove if you say, i don't mind. :p

**Comment by Akianonymus** [comment]
Anyway, do not merge now, need to push some fixes.
